src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/ProvideStreamMemoryBasedAsyncOverrides.cs

```diff
@@ -0,0 +1,208 @@
+            {
+                if (method.Parameters.Length != argumentTypes.Length)
+                    return false;
+
+                for (int index = 0; index < argumentTypes.Length; ++index)
+                {
+                    if (!argumentTypes[index].Equals(method.Parameters[index].Type, SymbolEqualityComparer.Default))
+                        return false;
+                }
+
+                return true;
+            }
+        }
+
+        //  If the specified type overrides the specified method on its immediate base class, returns the method symbol representing the 
+        //  overriding method. Returns null if the specified type does not override the specified method.
+        private static IMethodSymbol? GetSymbolForOverridingMethod(ITypeSymbol derivedType, IMethodSymbol overriddenMethod)
+        {
+            RoslynDebug.Assert(derivedType.BaseType.Equals(overriddenMethod.ContainingType, SymbolEqualityComparer.Default));
+
+            return derivedType.GetMembers(overriddenMethod.Name)
+                .SingleOrDefault(x =>
+                {
+                    return x.IsOverride && overriddenMethod.Equals(x.GetOverriddenMember(), SymbolEqualityComparer.Default);
+                }) as IMethodSymbol;
+        }
+
+        //  We will not be doing any comparisons on this type.
+#pragma warning disable CA1815 // Override equals and operator equals on value types
+        private readonly struct RequiredSymbols
+#pragma warning restore CA1815 // Override equals and operator equals on value types
+        {
+            public RequiredSymbols(
+                ITypeSymbol streamType, ITypeSymbol memoryOfByteType, ITypeSymbol readOnlyMemoryOfByteType,
+                IMethodSymbol readAsyncArrayMethod, IMethodSymbol readAsyncMemoryMethod,
+                IMethodSymbol writeAsyncArrayMethod, IMethodSymbol writeAsyncMemoryMethod)
+            {
+                StreamType = streamType;
+                MemoryOfByteType = memoryOfByteType;
+                ReadOnlyMemoryOfByteType = readOnlyMemoryOfByteType;
+                ReadAsyncArrayMethod = readAsyncArrayMethod;
+                ReadAsyncMemoryMethod = readAsyncMemoryMethod;
+                WriteAsyncArrayMethod = writeAsyncArrayMethod;
+                WriteAsyncMemoryMethod = writeAsyncMemoryMethod;
+            }
+
+            public ITypeSymbol StreamType { get; }
+            public ITypeSymbol MemoryOfByteType { get; }
+            public ITypeSymbol ReadOnlyMemoryOfByteType { get; }
+            public IMethodSymbol ReadAsyncArrayMethod { get; }
+            public IMethodSymbol ReadAsyncMemoryMethod { get; }
+            public IMethodSymbol WriteAsyncArrayMethod { get; }
+            public IMethodSymbol WriteAsyncMemoryMethod { get; }
+        }
+
+        private static LocalizableString CreateResource(string resourceName)
+        {
+            return new LocalizableResourceString(resourceName, Resx.ResourceManager, typeof(Resx));
```

Comment:
Should that be `MicrosoftNetCoreAnalyzersResources.ResourceManager` and `typeof(MicrosoftNetCoreAnalyzersResources)`?

Reply:
I placed a `using Resx = Microsoft.NetCore.Analyzers.MicrosoftNetCoreAnalyzersResources;` at the top of the file to reduce line-length in places.

Reply:
@NewellClark Ah thanks for clarifying. I didn't notice that line.